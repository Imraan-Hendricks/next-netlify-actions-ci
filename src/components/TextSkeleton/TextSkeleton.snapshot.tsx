import { render } from '@testing-library/react';
import { TextSkeleton } from './TextSkeleton';

describe('TextSkeleton snapshot tests', () => {
  it('renders with specified custom className', () => {
    const mathRandomSpy = jest.spyOn(Math, 'random').mockReturnValue(95);
    const { container } = render(<TextSkeleton className='custom-class' />);
    expect(container).toMatchSnapshot();
    mathRandomSpy.mockRestore();
  });
});
