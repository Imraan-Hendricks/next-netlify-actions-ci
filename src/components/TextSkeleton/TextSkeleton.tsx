import { ClassValue, cn } from '@/utils/common-utils';

interface TextSkeletonProps {
  className?: ClassValue;
}

export function TextSkeleton({ className }: TextSkeletonProps) {
  const min = 90;
  const max = 100;
  const num = Math.floor(Math.random() * (max - min + 1)) + min;
  let maxWidth: string;
  switch (num) {
    case 90:
      maxWidth = 'max-w-[90%]';
      break;
    case 91:
      maxWidth = 'max-w-[91%]';
      break;
    case 92:
      maxWidth = 'max-w-[92%]';
      break;
    case 93:
      maxWidth = 'max-w-[93%]';
      break;
    case 94:
      maxWidth = 'max-w-[94%]';
      break;
    case 95:
      maxWidth = 'max-w-[95%]';
      break;
    case 96:
      maxWidth = 'max-w-[96%]';
      break;
    case 97:
      maxWidth = 'max-w-[97%]';
      break;
    case 98:
      maxWidth = 'max-w-[98%]';
      break;
    case 99:
      maxWidth = 'max-w-[99%]';
      break;
    default:
      maxWidth = 'max-w-[100%]';
      break;
  }

  return (
    <div
      role='status'
      className={cn(
        'h-2 bg-gray-200 rounded-full animate-pulse',
        maxWidth,
        className,
      )}></div>
  );
}
