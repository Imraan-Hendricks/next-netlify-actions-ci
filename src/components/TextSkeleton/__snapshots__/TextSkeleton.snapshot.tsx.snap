// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSkeleton snapshot tests renders with specified custom className 1`] = `
<div>
  <div
    class="h-2 bg-gray-200 rounded-full animate-pulse max-w-[100%] custom-class"
    role="status"
  />
</div>
`;
